clients/batch-stats.js

```diff
@@ -0,0 +1,116 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var timers = require('timers');
+var os = require('os');
+var util = require('util');
+
+var BatchStatsd = require('tchannel/lib/statsd');
+
+module.exports = HyperbahnBatchStatsd;
+
+function HyperbahnBatchStatsd(opts) {
+    if (!(this instanceof HyperbahnBatchStatsd)) {
+        return new HyperbahnBatchStatsd(opts);
+    }
+
+    var self = this;
+
+    BatchStatsd.call(self, {
+        statsd: opts.statsd,
+        logger: opts.logger,
+        timers: timers,
+        baseTags: {
+            app: 'autobahn',
+            host: os.hostname()
+        }
+    });
+}
+util.inherits(HyperbahnBatchStatsd, BatchStatsd);
+
+HyperbahnBatchStatsd.prototype.handleStat =
+function handleStat(stat) {
+    /*eslint complexity: [2, 50]*/
+    var self = this;
+
+    if (!self.statsd) {
+        return;
+    } else if (isNullStatsd(self.statsd)) {
+        self.writeToStatsd(self.statsd, stat);
+        return;
+    }
+
+    if (
+        stat.name === 'tchannel.inbound.calls.recvd' ||
+        stat.name === 'tchannel.inbound.request.size' ||
+        stat.name === 'tchannel.inbound.request.size' ||
+        stat.name === 'tchannel.inbound.calls.latency' ||
+        stat.name === 'tchannel.inbound.calls.system-errors' ||
+        stat.name === 'tchannel.inbound.response.size' ||
+        stat.name === 'tchannel.inbound.calls.latency' ||
+        stat.name === 'tchannel.inbound.calls.success' ||
+        stat.name === 'tchannel.inbound.calls.app-errors' ||
+        stat.name === 'tchannel.inbound.response.size'
+    ) {
+        self.writeToStatsd(self.statsd.globalClient, stat);
+    } else if (
+        stat.name === 'tchannel.outbound.request.size' ||
+        stat.name === 'tchannel.outbound.calls.sent' ||
+        stat.name === 'tchannel.outbound.request.size' ||
+        stat.name === 'tchannel.outbound.calls.per-attempt-latency' ||
+        stat.name === 'tchannel.outbound.calls.per-attempt.operational-errors' ||
+        stat.name === 'tchannel.outbound.calls.system-errors' ||
+        stat.name === 'tchannel.outbound.response.size' ||
+        stat.name === 'tchannel.outbound.calls.per-attempt-latency' ||
+        stat.name === 'tchannel.outbound.calls.success' ||
+        stat.name === 'tchannel.outbound.calls.per-attempt.app-errors' ||
+        stat.name === 'tchannel.outbound.response.size'
+    ) {
+        self.writeToStatsd(self.statsd.globalClient, stat);
```

Comment:
For all these string checks, wouldn't it be much faster to check if the stat is in a precomputed hash?  Would greatly reduce string comparisons.